cluster/saltbase/salt/kube-addons/kube-addons.sh

```diff
@@ -19,6 +19,39 @@
 # managed result is of that. Start everything below that directory.
 KUBECTL=/usr/local/bin/kubectl
 
+function create-kubernetesauth-secret() {
+  token=$1
+  username=$2
+  safe_username=$(tr -s ':_' '--' <<< "${username}")
+
+  # Make secret with a kubernetes_auth file with a token.
+  # TODO(etune): put apiserver certs into secret too, and reference from authfile,
+  # so that "Insecure" is not needed.
+  kafile=$(echo "{\"BearerToken\": \"${token}\", \"Insecure\": true }" | base64 -w0)
```

Comment:
```
local kafile
kafile=$( )
```

It's less important in the script as it stands because it currently violates our existing standards of having:
```
set -o errexit
set -o nounset
set -o pipefail
```

... but if `errexit` were set, the setting of the variable and the "local" need to be on separate lines. That's all I was saying.